module/platform/darwin/SensorFilter/src/SensorFilter.cpp

```diff
@@ -491,43 +514,37 @@ namespace module::platform::darwin {
                                              config.motionFilter.noise.measurement.flatFootOrientation,
                                              MeasurementType::FLAT_FOOT_ORIENTATION());
                     }
+                    // Else the foot is off the ground, so we make sure that for the next time step, we know that
+                    // this time step, the foot was off the ground
                     else if (!foot_down) {
                         previous_foot_down[side] = false;
                     }
 
+                    // Note that the Hwf is set, even if the foot is not down. This means that moving feet in the air
+                    // will have an Hwf associated with them which is the transform from when that foot last hit
+                    // the ground
                     sensors->feet[side].Hwf = footlanding_Hwf[side].matrix();
```

Comment:
We record it and send it because it makes debugging easier. The (still not quite complete) odometry view in NUsight2 shows the last foot landing position which makes it easier to see where it was when the foot was on the ground.